phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -581,92 +601,135 @@ private boolean areMutationsInSameTable(Table targetHTable, Region region) {
-                                            + "deleted cells for a table with disabled index: "
-                                            + fullTableName);
-                                    isDisabled = true;
-                                    break;
-                                }
-                            }
-                        }
-                        if (table != null && !isDisabled && isPhoenixTableTTLEnabled) {
-                            internalScanner =
-                                    new CompactionScanner(c.getEnvironment(), store, scanner,
-                                            MetaDataUtil.getMaxLookbackAge(
-                                                    c.getEnvironment().getConfiguration(), table.getMaxLookbackAge()),
-                                            SchemaUtil.getEmptyColumnFamily(table),
-                                            table.getEncodingScheme()
-                                                    == PTable.QualifierEncodingScheme.NON_ENCODED_QUALIFIERS ?
-                                                    QueryConstants.EMPTY_COLUMN_BYTES :
-                                                    table.getEncodingScheme().encode(QueryConstants.ENCODED_EMPTY_COLUMN_NAME)
-                                            );
+
+        final TableName tableName = c.getEnvironment().getRegion().getRegionInfo().getTable();
+        // Compaction and split upcalls run with the effective user context of the requesting user.
+        // This will lead to failure of cross cluster RPC if the effective user is not
+        // the login user. Switch to the login user context to ensure we have the expected
+        // security context.
+        return User.runAsLoginUser(new PrivilegedExceptionAction<InternalScanner>() {
+            @Override
+            public InternalScanner run() throws Exception {
+                InternalScanner internalScanner = scanner;
+                boolean keepDeleted = false;
+                boolean isMultiTenantIndexTable = false;
+                if (tableName.getNameAsString().startsWith(MetaDataUtil.VIEW_INDEX_TABLE_PREFIX)) {
+                    isMultiTenantIndexTable = true;
+                }
+                final String fullTableName = isMultiTenantIndexTable ?
+                        SchemaUtil.getParentTableNameFromIndexTable(tableName.getNameAsString(),
+                                MetaDataUtil.VIEW_INDEX_TABLE_PREFIX) :
+                        tableName.getNameAsString();
+                PTable table = null;
+                Integer maxLookbackAge = null;
+                try (PhoenixConnection conn = QueryUtil.getConnectionOnServer(
+                        compactionConfig).unwrap(PhoenixConnection.class)) {
+                    table = conn.getTableNoCache(fullTableName);
+                    maxLookbackAge = table.getMaxLookbackAge();
+                    if (table.getType() == PTableType.INDEX) {
```

Comment:
I think we don't need this `if` check now as Indexes inherit max lookback of tables and view indexes inherit max lookback of parent view?